diff --git a/node_modules/@uniswap/v3-core/contracts/libraries/TickMath.sol b/node_modules/@uniswap/v3-core/contracts/libraries/TickMath.sol
index 378e445..38c7505 100644
--- a/node_modules/@uniswap/v3-core/contracts/libraries/TickMath.sol
+++ b/node_modules/@uniswap/v3-core/contracts/libraries/TickMath.sol
@@ -22,7 +22,7 @@ library TickMath {
     /// at the given tick
     function getSqrtRatioAtTick(int24 tick) internal pure returns (uint160 sqrtPriceX96) {
         uint256 absTick = tick < 0 ? uint256(-int256(tick)) : uint256(int256(tick));
-        require(absTick <= uint256(MAX_TICK), 'T');
+        require(absTick <= uint256(887272), 'T');
 
         uint256 ratio = absTick & 0x1 != 0 ? 0xfffcb933bd6fad37aa2d162d1a594001 : 0x100000000000000000000000000000000;
         if (absTick & 0x2 != 0) ratio = (ratio * 0xfff97272373d413259a46990580e213a) >> 128;
